src/include/migraphx/eliminate_fp8.hpp

```diff
@@ -21,23 +21,32 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-#include <limits>
-#include "verify_program.hpp"
-#include <migraphx/program.hpp>
-#include <migraphx/generate.hpp>
-#include <migraphx/make_op.hpp>
-#include <migraphx/half.hpp>
+#ifndef MIGRAPHX_GUARD_AMDMIGRAPHX_ELIMINATE_FP8_HPP
+#define MIGRAPHX_GUARD_AMDMIGRAPHX_ELIMINATE_FP8_HPP
 
-struct test_isnan_half : verify_program<test_isnan_half>
+#include <migraphx/config.hpp>
+#include <migraphx/shape.hpp>
+#include <set>
+#include <string>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+struct module;
+
+/**
+This will insert convert operators for the operators that are not implemented for FP8 dtypes
+ */
+struct MIGRAPHX_EXPORT eliminate_fp8
 {
-    migraphx::program create_program() const
-    {
-        migraphx::program p;
-        auto* mm = p.get_main_module();
-        auto x   = mm->add_parameter("x", migraphx::shape{migraphx::shape::half_type, {2}});
-        auto l0  = mm->add_literal(std::numeric_limits<migraphx::half>::quiet_NaN());
-        x        = mm->add_instruction(migraphx::make_op("concat", {{"axis", 0}}), x, l0);
-        mm->add_instruction(migraphx::make_op("isnan"), x);
-        return p;
-    }
+    // TODO: Add all device ops as a later PR and add tests for those.
+    std::set<std::string> op_names;
```

Comment:
I think you could just add `op_names` to `eliminate_data_type` pass. If its empty than apply to all operators otherwise fallback on default behavior.

Reply:
I need to run this pass after auto_contiguous. Otherwise contiguous's with fp8 shows up in some tests.

Reply:
Yes, I am saying run a additional `eliminate_data_type` in the same place you are running `eliminate_fp8` currently. No need to write a new pass.

Reply:
okay, eliminated `eliminate_fp8` pass using `eliminate_data_type` pass.